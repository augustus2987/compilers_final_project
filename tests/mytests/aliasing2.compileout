compiler:: base tree:
compiler::	Assign([AssName('x', 'OP_ASSIGN')], Dict([(Const(5), Const(15)), (Const(6), Const(16)), (Const(7), Const(17)), (Const(8), Name('True'))]))
compiler::	Assign([AssName('y', 'OP_ASSIGN')], Name('x'))
compiler::	Assign([Subscript(Name('x'), 'OP_ASSIGN', [Const(5)])], Name('False'))
compiler::	Printnl([Name('x')], None)
explicate:: Explicated Tree:
Module(None, Stmt([Assign([AssName('x', 'OP_ASSIGN')], Dict([(Const(5), Const(15)), (Const(6), Const(16)), (Const(7), Const(17)), (Const(8), Bool(True))])), Assign([AssName('y', 'OP_ASSIGN')], Name('x')), Assign([Subscript(Name('x'), 'OP_ASSIGN', [Const(5)])], Bool(False)), Let(Name('expl_0'), Name('x'), IfExp(InjectFrom(bool, Compare(Const(0), [('==', GetTag(Name('expl_0')))])), Printnl([ProjectTo(int, Name('expl_0'))], None), IfExp(InjectFrom(bool, Compare(Const(1), [('==', GetTag(Name('expl_0')))])), PrintBool(Name('expl_0')), IfExp(InjectFrom(bool, Compare(Const(3), [('==', GetTag(Name('expl_0')))])), PrintBig(Name('expl_0')), CallFunc(Name('call_error'), [], None, None)))))]))
Flatten:: Flattened AST tree:
Flatten:: 	Assign([AssName('var_0', 'OP-ASSIGN')], Dict([(Const(5), Const(15)), (Const(6), Const(16)), (Const(7), Const(17)), (Const(8), Bool(True))]))
Flatten:: 	Assign([AssName('user_x', 'OP_ASSIGN')], Name('var_0'))
Flatten:: 	Assign([AssName('user_y', 'OP_ASSIGN')], Name('user_x'))
Flatten:: 	Assign([Subscript(Name('user_x'), 'OP_ASSIGN', [Const(5)])], Bool(False))
Flatten:: 	Let(Name('expl_0'), Name('user_x'), Stmt([Assign([AssName('var_19', 'OP-ASSIGN')], GetTag(Name('expl_0'))), Assign([AssName('var_18', 'OP-ASSIGN')], Compare(Const(0), [('==', Name('var_19'))])), Assign([AssName('var_17', 'OP-ASSIGN')], InjectFrom(bool, Name('var_18'))), Assign([AssName('var_22', 'OP-ASSIGN')], GetTag(Name('var_17'))), Assign([AssName('var_22', 'OP-ASSIGN')], Compare(Const(0), [('==', Name('var_22'))])), IfExp(Name('var_22'), Stmt([Assign([AssName('var_17', 'OP-ASSIGN')], ProjectTo(int, Name('var_17')))]), Stmt([Assign([AssName('var_17', 'OP-ASSIGN')], ProjectTo(bool, Name('var_17')))])), IfExp(Name('var_17'), Stmt([Assign([AssName('var_1', 'OP-ASSIGN')], ProjectTo(int, Name('expl_0'))), Printnl([Name('var_1')], None)]), Stmt([Assign([AssName('var_12', 'OP-ASSIGN')], GetTag(Name('expl_0'))), Assign([AssName('var_11', 'OP-ASSIGN')], Compare(Const(1), [('==', Name('var_12'))])), Assign([AssName('var_10', 'OP-ASSIGN')], InjectFrom(bool, Name('var_11'))), Assign([AssName('var_15', 'OP-ASSIGN')], GetTag(Name('var_10'))), Assign([AssName('var_15', 'OP-ASSIGN')], Compare(Const(0), [('==', Name('var_15'))])), IfExp(Name('var_15'), Stmt([Assign([AssName('var_10', 'OP-ASSIGN')], ProjectTo(int, Name('var_10')))]), Stmt([Assign([AssName('var_10', 'OP-ASSIGN')], ProjectTo(bool, Name('var_10')))])), IfExp(Name('var_10'), Stmt([PrintBool(Name('expl_0'))]), Stmt([Assign([AssName('var_5', 'OP-ASSIGN')], GetTag(Name('expl_0'))), Assign([AssName('var_4', 'OP-ASSIGN')], Compare(Const(3), [('==', Name('var_5'))])), Assign([AssName('var_3', 'OP-ASSIGN')], InjectFrom(bool, Name('var_4'))), Assign([AssName('var_8', 'OP-ASSIGN')], GetTag(Name('var_3'))), Assign([AssName('var_8', 'OP-ASSIGN')], Compare(Const(0), [('==', Name('var_8'))])), IfExp(Name('var_8'), Stmt([Assign([AssName('var_3', 'OP-ASSIGN')], ProjectTo(int, Name('var_3')))]), Stmt([Assign([AssName('var_3', 'OP-ASSIGN')], ProjectTo(bool, Name('var_3')))])), IfExp(Name('var_3'), Stmt([PrintBig(Name('expl_0'))]), Stmt([Assign([AssName('var_2', 'OP-ASSIGN')], CallFunc(Name('call_error'), [], None, None)), Assign([AssName('var_2', 'OP-ASSIGN')], Name('var_2')), Assign([AssName('var_2', 'OP-ASSIGN')], Name('var_2'))]))]))]))]))
x86IR::  Intermediate representation:
x86IR::  	call create_dict
x86IR::  	pushl eax
x86IR::  	call inject_big
x86IR::  	movl eax var_0
x86IR::  	pushl $5
x86IR::  	call inject_int
x86IR::  	movl eax ebx
x86IR::  	pushl $15
x86IR::  	call inject_int
x86IR::  	pushl eax
x86IR::  	pushl ebx
x86IR::  	pushl var_0
x86IR::  	call set_subscript
x86IR::  	pushl $6
x86IR::  	call inject_int
x86IR::  	movl eax ebx
x86IR::  	pushl $16
x86IR::  	call inject_int
x86IR::  	pushl eax
x86IR::  	pushl ebx
x86IR::  	pushl var_0
x86IR::  	call set_subscript
x86IR::  	pushl $7
x86IR::  	call inject_int
x86IR::  	movl eax ebx
x86IR::  	pushl $17
x86IR::  	call inject_int
x86IR::  	pushl eax
x86IR::  	pushl ebx
x86IR::  	pushl var_0
x86IR::  	call set_subscript
x86IR::  	pushl $8
x86IR::  	call inject_int
x86IR::  	movl eax ebx
x86IR::  	pushl $1
x86IR::  	call inject_bool
x86IR::  	pushl eax
x86IR::  	pushl ebx
x86IR::  	pushl var_0
x86IR::  	call set_subscript
x86IR::  	movl var_0 user_x
x86IR::  	movl user_x user_y
x86IR::  	pushl $0
x86IR::  	call inject_bool
x86IR::  	movl eax edx
x86IR::  	pushl $5
x86IR::  	call inject_int
x86IR::  	pushl edx
x86IR::  	pushl eax
x86IR::  	pushl user_x
x86IR::  	call set_subscript
x86IR::  	movl user_x expl_0
x86IR::  	pushl expl_0
x86IR::  	call tag
x86IR::  	movl eax var_19
x86IR::  	cmpl $0 var_19
x86IR::  	je ir_tag0_cmp
x86IR::  	movl $0 var_18
x86IR::  	jmp ir_tag0_cmpend
x86IR::  	ir_tag0_cmp:
x86IR::  	movl $1 var_18
x86IR::  	ir_tag0_cmpend:
x86IR::  	pushl var_18
x86IR::  	call inject_bool
x86IR::  	movl eax var_17
x86IR::  	pushl var_17
x86IR::  	call tag
x86IR::  	movl eax var_22
x86IR::  	cmpl $0 var_22
x86IR::  	je ir_tag1_cmp
x86IR::  	movl $0 var_22
x86IR::  	jmp ir_tag1_cmpend
x86IR::  	ir_tag1_cmp:
x86IR::  	movl $1 var_22
x86IR::  	ir_tag1_cmpend:
x86IR::  	cmpl $0 var_22
x86IR::  	je ir_tag_else_2
x86IR::  	pushl var_17
x86IR::  	call project_int
x86IR::  	movl eax var_17
x86IR::  	jmp ir_tag_end_2
x86IR::  	ir_tag_else_2:
x86IR::  	pushl var_17
x86IR::  	call project_bool
x86IR::  	movl eax var_17
x86IR::  	jmp ir_tag_end_2
x86IR::  	ir_tag_end_2:
x86IR::  	cmpl $0 var_17
x86IR::  	je ir_tag_else_3
x86IR::  	pushl expl_0
x86IR::  	call project_int
x86IR::  	movl eax var_1
x86IR::  	pushl var_1
x86IR::  	call print_int_nl
x86IR::  	jmp ir_tag_end_3
x86IR::  	ir_tag_else_3:
x86IR::  	pushl expl_0
x86IR::  	call tag
x86IR::  	movl eax var_12
x86IR::  	cmpl $1 var_12
x86IR::  	je ir_tag4_cmp
x86IR::  	movl $0 var_11
x86IR::  	jmp ir_tag4_cmpend
x86IR::  	ir_tag4_cmp:
x86IR::  	movl $1 var_11
x86IR::  	ir_tag4_cmpend:
x86IR::  	pushl var_11
x86IR::  	call inject_bool
x86IR::  	movl eax var_10
x86IR::  	pushl var_10
x86IR::  	call tag
x86IR::  	movl eax var_15
x86IR::  	cmpl $0 var_15
x86IR::  	je ir_tag5_cmp
x86IR::  	movl $0 var_15
x86IR::  	jmp ir_tag5_cmpend
x86IR::  	ir_tag5_cmp:
x86IR::  	movl $1 var_15
x86IR::  	ir_tag5_cmpend:
x86IR::  	cmpl $0 var_15
x86IR::  	je ir_tag_else_6
x86IR::  	pushl var_10
x86IR::  	call project_int
x86IR::  	movl eax var_10
x86IR::  	jmp ir_tag_end_6
x86IR::  	ir_tag_else_6:
x86IR::  	pushl var_10
x86IR::  	call project_bool
x86IR::  	movl eax var_10
x86IR::  	jmp ir_tag_end_6
x86IR::  	ir_tag_end_6:
x86IR::  	cmpl $0 var_10
x86IR::  	je ir_tag_else_7
x86IR::  	pushl expl_0
x86IR::  	call print_any
x86IR::  	jmp ir_tag_end_7
x86IR::  	ir_tag_else_7:
x86IR::  	pushl expl_0
x86IR::  	call tag
x86IR::  	movl eax var_5
x86IR::  	cmpl $3 var_5
x86IR::  	je ir_tag8_cmp
x86IR::  	movl $0 var_4
x86IR::  	jmp ir_tag8_cmpend
x86IR::  	ir_tag8_cmp:
x86IR::  	movl $1 var_4
x86IR::  	ir_tag8_cmpend:
x86IR::  	pushl var_4
x86IR::  	call inject_bool
x86IR::  	movl eax var_3
x86IR::  	pushl var_3
x86IR::  	call tag
x86IR::  	movl eax var_8
x86IR::  	cmpl $0 var_8
x86IR::  	je ir_tag9_cmp
x86IR::  	movl $0 var_8
x86IR::  	jmp ir_tag9_cmpend
x86IR::  	ir_tag9_cmp:
x86IR::  	movl $1 var_8
x86IR::  	ir_tag9_cmpend:
x86IR::  	cmpl $0 var_8
x86IR::  	je ir_tag_else_10
x86IR::  	pushl var_3
x86IR::  	call project_int
x86IR::  	movl eax var_3
x86IR::  	jmp ir_tag_end_10
x86IR::  	ir_tag_else_10:
x86IR::  	pushl var_3
x86IR::  	call project_bool
x86IR::  	movl eax var_3
x86IR::  	jmp ir_tag_end_10
x86IR::  	ir_tag_end_10:
x86IR::  	cmpl $0 var_3
x86IR::  	je ir_tag_else_11
x86IR::  	pushl expl_0
x86IR::  	call print_any
x86IR::  	jmp ir_tag_end_11
x86IR::  	ir_tag_else_11:
x86IR::  	call call_error
x86IR::  	pushl eax
x86IR::  	call inject_int
x86IR::  	movl eax var_2
x86IR::  	movl var_2 var_2
x86IR::  	movl var_2 var_2
x86IR::  	jmp ir_tag_end_11
x86IR::  	ir_tag_end_11:
x86IR::  	jmp ir_tag_end_7
x86IR::  	ir_tag_end_7:
x86IR::  	jmp ir_tag_end_3
x86IR::  	ir_tag_end_3:
VariableGraph::  # of variables: 20
