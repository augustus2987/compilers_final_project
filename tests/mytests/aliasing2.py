x = {5: 15, 6:16, 7:17, 8:True}
y = x

x[5] = False
print x
