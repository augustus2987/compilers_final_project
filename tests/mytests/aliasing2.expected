{8: True, 5: False, 6: 16, 7: 17}
